### ISSUE TITLE
Unclear from docs why `sim.samples` of a single handle produces array of datasets

### ISSUE THREAD

**Issue by Helveg** [arbor-sim/arbor#1877]
The docs here https://docs.arbor-sim.org/en/latest/python/simulation.html#recording-samples explain it as:

> and returns the recorded data as a list, with one entry for **each probe** associated with the **probe id** that was used in step 2 above

But all the text above mentions probes and probe ids as a singular thing, produced by an index entry in the return array of `recipe.probes`; so why does my `handle = sim.sample(probe_id, ...)` call, with all singular references to probe and handle suddenly produce multiple datasets?

Answer: Because each `probe(s?)` can be expanded into multiple recording sites because the `locset` can point to multiple locations. 

That isn't really covered in the doc there. And makes the `data, meta = sim.samples(handle)[0]` susceptible to become cargo cult pattern: Don't know why, but I copy pasted this from the docs.

**Comment by brenthuisman** [comment]
Probe-ids being tied to a locset, and not a singular location, is a 'recent' feature and the preceding text was clearly written before it was merged and not updated. I'll review this and make changes where necessary. 

I'm not sure much can be done about `data, meta = sim.samples(handle)[0]` becoming a recurring pattern; setting up an id with a multi-position locset is perhaps a bit niche (for now). We could auto-unpack, but then you'll have the opposite problem when you _do_ use a multi-position locset: `why is my code failing? Ah, because I forgot I must unpack now.`

**Comment by brenthuisman** [comment]
Actually, reading the text above it is mentioned more than once that a `probe id` is associated with a locset and thus may describe more than one location, and that therefore a list is returned. It's true that it is not written every time probes are mentioned, but that seems like it would be too redundant. So, I'm not sure where it becomes unclear to you; at your quote it is again elaborated. Can you maybe point at a few specific lines where you can't make sense of it?

**Comment by Helveg** [comment]
The usage of "probe id" and "probe adress" often coincides, and both have a 0 to n component, making the text hard to follow.

eg 

> probe id: A designator for one or more probes as specified by a recipe

but then also

> Probes are described by probe addresses, and the collection of probe addresses for a given cell is provided by the [recipe](https://docs.arbor-sim.org/en/latest/python/recipe.html#arbor.recipe) object. One address may correspond to more than one probe:

This implies that both probe ids and probe adresses describe one or more probes? One of the two explanations must be inacurate. I think a probe id is a designator for exactly one _probe address_, where an address can contain one or more probes

**Comment by brenthuisman** [comment]
Yeah, `probeset` and then `probeset id` (and not address) would be very consistent :) I'll shove it in the dev/doc PR.

Re handles: there's a PR for that: https://github.com/arbor-sim/arbor/issues/1539
